File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/MyVideoItemFactory.java
package me.xiaopan.sketchsample.adapter.itemfactory;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory;
import me.xiaopan.sketchsample.ImageOptions;
import me.xiaopan.sketchsample.R;
import me.xiaopan.sketchsample.adapter.BindAssemblyRecyclerItem;
import me.xiaopan.sketchsample.bean.VideoItem;
import me.xiaopan.sketchsample.util.VideoThumbnailPreprocessor;
import me.xiaopan.sketchsample.widget.SampleImageView;

public class MyVideoItemFactory extends AssemblyRecyclerItemFactory<MyVideoItemFactory.MyVideoItem> {
    private MyVideoItemListener listener;

    public MyVideoItemFactory(MyVideoItemListener listener) {
        this.listener = listener;
    }

    @Override
    public boolean isTarget(Object o) {
        return o instanceof VideoItem;
    }

    @Override
    public MyVideoItem createAssemblyItem(ViewGroup viewGroup) {
        return new MyVideoItem(R.layout.list_item_my_video, viewGroup);
    }

    public interface MyVideoItemListener {
        void onClickVideo(int position, VideoItem videoItem);
    }

    public class MyVideoItem extends BindAssemblyRecyclerItem<VideoItem> {
        @BindView(R.id.image_myVideoItem_icon)
        SampleImageView iconImageView;

        @BindView(R.id.text_myVideoItem_name)
        TextView nameTextView;

        @BindView(R.id.text_myVideoItem_size)
        TextView sizeTextView;

        @BindView(R.id.text_myVideoItem_date)
        TextView dateTextView;

        @BindView(R.id.text_myVideoItem_duration)
        TextView durationTextView;

        public MyVideoItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onConfigViews(Context context) {
            iconImageView.setOptions(ImageOptions.RECT);
            iconImageView.setPage(SampleImageView.Page.PHOTO_LIST);

            getItemView().setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onClickVideo(getPosition(), getData());
                    }
                }
            });

            iconImageView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    getItemView().performClick();
                }
            });
        }

        @Override
        protected void onSetData(int i, VideoItem videoItem) {
            iconImageView.displayImage(VideoThumbnailPreprocessor.createUri(videoItem.path));
            nameTextView.setText(videoItem.title);
            sizeTextView.setText(videoItem.getTempFormattedSize(sizeTextView.getContext()));
            dateTextView.setText(String.valueOf(videoItem.getTempFormattedDate()));
            durationTextView.setText(String.valueOf(videoItem.getTempFormattedDuration()));
        }
    }
}

File: sample-video-thumbnail/src/main/java/me/panpf/sketch/sample/videothumbnail/MyVideoItemFactory.kt
package me.panpf.sketch.sample.videothumbnail

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import me.xiaopan.assemblyadapter.AssemblyRecyclerItem
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory
import me.panpf.sketch.SketchImageView
import me.panpf.sketch.display.TransitionImageDisplayer

class MyVideoItemFactory(private val listener: MyVideoItemListener?) : AssemblyRecyclerItemFactory<MyVideoItemFactory.MyVideoItem>() {

    override fun isTarget(o: Any): Boolean {
        return o is VideoItem
    }

    override fun createAssemblyItem(viewGroup: ViewGroup): MyVideoItem {
        return MyVideoItem(R.layout.list_item_my_video, viewGroup)
    }

    interface MyVideoItemListener {
        fun onClickVideo(position: Int, videoItem: VideoItem)
    }

    inner class MyVideoItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyRecyclerItem<VideoItem>(itemLayoutId, parent) {
        val iconImageView: SketchImageView by bindView(R.id.image_myVideoItem_icon)
        val nameTextView: TextView by bindView(R.id.text_myVideoItem_name)
        val sizeTextView: TextView by bindView(R.id.text_myVideoItem_size)
        val dateTextView: TextView by bindView(R.id.text_myVideoItem_date)
        val durationTextView: TextView by bindView(R.id.text_myVideoItem_duration)

        override fun onConfigViews(context: Context) {
            getItemView().setOnClickListener {
                listener?.onClickVideo(position, data)
            }

            iconImageView.onClickListener = View.OnClickListener { getItemView().performClick() }

            iconImageView.options
                    .setLoadingImage(R.drawable.image_loading).displayer = TransitionImageDisplayer()
        }

        override fun onSetData(i: Int, videoItem: VideoItem) {
            iconImageView.displayImage(VideoThumbnailUriModel.makeUri(videoItem.path ?: ""))
            nameTextView.text = videoItem.title
            sizeTextView.text = videoItem.getTempFormattedSize(sizeTextView.context)
            dateTextView.text = videoItem.tempFormattedDate
            durationTextView.text = videoItem.tempFormattedDuration
        }
    }
}
